src/main/scala/org/apache/spark/shuffle/ucx/ShuffleTransport.scala

```diff
@@ -25,6 +26,9 @@ trait BlockId
 trait Block {
   // Transport will call this method when it would need an actual block memory.
   def getMemoryBlock: MemoryBlock
+
+  // Private transport lock to know when there are outstanding operations to block memory.
+  private[ucx] val lock = new StampedLock().asReadWriteLock()
```

Comment:
It's private to ucx package only. In outside packages you won't see it

Reply:
it would be better if it wasn't exposed in the API, and if code didn't reach into block and call methods on a member of `Block`. At least there should be an interface for this (methods), and I think that interface (and the lock) being in a wrapper of `Block` that is internal to the transport would make it cleaner. That said, I think this can be cleaned up later.